### PR TITLE
Bug/Kyber app can't detect account [#4410]

### LINKED ISSUE #4410: Kyber app can't detect account
### User Story

As a user, I want to log in to Kyber with my Status wallet.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

*Summary*:
When logging in on Kyber, the app throws an error saying it cannot detect a Metamask account.

#### Expected behavior
Kyber detects the user's account.

### Reproduction

- Open Kyber from DApp screen
- .Accept Terms and Conditions
- "Get Started"

### Additional Information
* Status version: nightly 2018-05-24
* Operating System: iOS

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #4410

### CODE REVIEW

File: resources/js/web3_init.js [hunk 1]
```diff
@@ -53,18 +53,23 @@ StatusHttpProvider.prototype.prepareRequest = function () {
...
-    var messageId = callbackId++;
-    callbacks[messageId] = callback;
-    if (typeof StatusBridge == "undefined") {
-        var data = {
-            payload: JSON.stringify(payload),
-            callbackId: JSON.stringify(messageId),
-            host: this.host
-        };
+    if (payload.method == "eth_coinbase"){
+        callback(null, web3Response(payload, currentAccountAddress));
```
Comment (flexsurfer) [inline resources/js/web3_init.js]: added callback for async "eth_coinbase"

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/4xkpv6 for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/4xkpv6">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/rtB6CW for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/rtB6CW">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by jeluard — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/cbnM1o for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/cbnM1o">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/HtgeXx for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/HtgeXx">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/m1qefJ for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/m1qefJ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: resources/js/web3_init.js (hunk 1 above)
Comment (mandrigin) [reply resources/js/web3_init.js]: can we do the same for `eth_accounts` and `net_version` to be consistent with sync methods?

**Review by mandrigin — APPROVED** [review]

File: resources/js/web3_init.js (hunk 1 above)
Comment (flexsurfer) [reply resources/js/web3_init.js]: sounds reasonable @mandrigin

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/twQoZa for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/twQoZa">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/3t7KfE for branch PR-4469

<blockquote><div><strong><a href="https://i.diawi.com/3t7KfE">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Pull request merged**